cirq-core/cirq/optimizers/two_qubit_to_sqrt_iswap_test.py

```diff
@@ -0,0 +1,402 @@
+    *perturbations_weyl(1 / 2 * np.pi / 4, 1 / 2 * np.pi / 8, 1 / 2 * 0),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * 0, 1 / 4 * 0),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi / 4, 1 / 4 * 0),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi / 8, 1 / 4 * np.pi / 8),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi / 8, 1 / 4 * -np.pi / 8),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi * 1 / 16, 1 / 4 * np.pi / 16),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi * 1 / 16, 1 / 4 * -np.pi / 16),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi * 3 / 16, 1 / 4 * np.pi / 16),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi * 3 / 16, 1 / 4 * -np.pi / 16),
+    *perturbations_weyl(1 / 4 * np.pi / 4, 1 / 4 * np.pi / 8, 1 / 4 * 0),
+    # Some random points
+    random_unitary(17323),
+    random_unitary(99415),
+    random_unitary(65832),
+    random_unitary(45373),
+    random_unitary(27123),
+]
+ONE_SQRT_ISWAP_UNITARIES = [
+    *perturbations_gate(cirq.SQRT_ISWAP),
+    *perturbations_gate(cirq.ISwapPowGate(exponent=-0.5, global_shift=1)),
+    *perturbations_unitary(random_locals(np.pi / 8, np.pi / 8, 0, seed=80342)),
+    *perturbations_unitary(random_locals(np.pi / 8, np.pi / 8, 0, seed=83551)),
+    *perturbations_unitary(random_locals(np.pi / 8, np.pi / 8, 0, seed=52450)),
+]
+ZERO_UNITARIES = [
+    *perturbations_gate(cirq.IdentityGate(2)),
+    *perturbations_unitary(random_locals(0, 0, 0, seed=62933)),
+    *perturbations_unitary(random_locals(0, 0, 0, seed=50590)),
+]
+
+# Lists of seeds for cirq.testing.random_unitary(4, random_state=seed)
+# corresponding to different regions in the Weyl chamber and possible
+# eigenvalue crossings in the 3-SQRT_ISWAP decomposition
+# Key: 1 (fig. 4):  x < y + abs(z)
+#      2 (fig. 7a): x > pi/8
+#      4 (fig. 7b): z < 0
+#      8 (fig. 7c): y+abs(z) < pi/8 if x <= pi/8 else y+abs(z) < pi/4
+WEYL_REGION_UNITARIES = {
+    1: list(map(random_unitary, [49903, 84819, 84769])),
+    3: list(map(random_unitary, [89483, 79607, 96012])),
+    5: list(map(random_unitary, [31979, 78758, 47661])),
+    7: list(map(random_unitary, [46429, 80635, 49695])),
+    8: list(map(random_unitary, [86872, 91499, 47843])),
+    9: list(map(random_unitary, [48333, 48333, 95468])),
+    10: list(map(random_unitary, [97633, 99062, 44397])),
+    11: list(map(random_unitary, [42319, 41651, 11322])),
+    12: list(map(random_unitary, [48530, 64117, 31880])),
+    13: list(map(random_unitary, [95140, 13781, 60879])),
+    14: list(map(random_unitary, [85837, 14090, 73607])),
+    15: list(map(random_unitary, [67955, 86488, 24698])),
+}
+ALL_REGION_UNITARIES = list(itertools.chain(*WEYL_REGION_UNITARIES.values()))
+
+
+def assert_valid_decomp(
+    u_target,
+    operations,
+    *,
+    single_qubit_gate_types=(cirq.ZPowGate, cirq.XPowGate, cirq.YPowGate),
+    two_qubit_gate=cirq.SQRT_ISWAP,
+    atol=1e-6,
```

Comment:
It would be nice if the default `atol` in `cirq.allclose_up_to_global_phase` is enough. Right now, the default `atol` returns false in many cases.

Reply:
Agreed.  There is some source of numerical error that I couldn't track down.  It doesn't help that some the Cirq linalg functions don't pass through atol to half of the functions they use internally (so default to 1e-8).

Reply:
Added a comment.  It may be worth looking into the cause of this and also fixing various linalg functions that don't pass atol to their sub-functions.